python_modules/dagster/dagster/daemon/sensor.py

```diff
@@ -169,6 +170,56 @@ def _check_for_debug_crash(debug_crash_flags, key):
 RELOAD_WORKSPACE = 60
 
 
+class SensorFutureContext(
+    NamedTuple(
+        "_SensorFutureContext",
+        [
+            ("future", concurrent.futures.Future),
+            ("timeout", float),
+        ],
+    )
+):
+    DEFAULT_TIMEOUT_SECONDS = 75
+
+    def __new__(
+        cls,
+        future: concurrent.futures.Future,
+        timeout: Optional[float] = None,
+    ):
+        return super(cls, SensorFutureContext).__new__(
+            cls,
+            check.inst_param(future, "future", concurrent.futures.Future),
+            check.opt_float_param(
+                timeout,
+                "timeout",
+                default=pendulum.now("UTC").add(seconds=cls.DEFAULT_TIMEOUT_SECONDS).timestamp(),
```

Comment:
realizing that this needs some more thought when used in this context... if we're not careful and things fall behind, you could get to a point where every future is timing out